src/main/java/seedu/address/model/person/Email.java

```diff
@@ -48,7 +49,8 @@ public Email(String email) {
      * Returns if a given string is a valid email.
      */
     public static boolean isValidEmail(String test) {
-        return test.matches(VALIDATION_REGEX);
+        String[] emailArray = test.split("");
+        return ((emailArray.length <= 50) && test.matches(VALIDATION_REGEX));
```

Comment:
Good implementation of the check of for email length less than equal to 50 as proposed in our meeting :100: